ocs_ci/ocs/node.py

```diff
@@ -1107,3 +1107,63 @@ def is_node_labeled(node_name, label=constants.OPERATOR_NODE_LABEL):
     """
     node_names_with_label = machine.get_labeled_nodes(label=label)
     return node_name in node_names_with_label
+
+
+def check_taint_on_nodes(taint=None):
+    """
+    Function to check for particular taint on nodes
+
+    Args:
+        taint (str): The taint to check on nodes
+
+    """
+    taint = taint if taint else constants.OPERATOR_NODE_TAINT
+    ocs_nodes = get_ocs_nodes()
+    flag = -1
+    for node_obj in ocs_nodes:
+        if node_obj.get().get("spec").get("taints"):
+            if taint in node_obj.get().get("spec").get("taints")[0].get("key"):
+                log.info(f"Node {node_obj.name} has taint {taint}")
+                flag = 1
+        else:
+            flag = 0
+        return bool(flag)
+
+
+def taint_nodes(nodes_to_taint=None):
+    """
+    Function to taint nodes
+
+    Args:
+        nodes_to_taint (list): Nodes to taint
+
+    """
+    if not check_taint_on_nodes():
+        ocp = OCP()
+        ocs_nodes = get_ocs_nodes()
+        nodes_to_taint = nodes_to_taint if nodes_to_taint else ocs_nodes
+        log.info(f"Taint nodes with taint: " f"{constants.OPERATOR_NODE_TAINT}")
+        for node in nodes_to_taint:
+            taint_cmd = f"adm taint nodes {node.name} {constants.OPERATOR_NODE_TAINT}"
+            ocp.exec_oc_cmd(command=taint_cmd)
+    else:
+        log.info(f"Nodes already have taint {constants.OPERATOR_NODE_TAINT} ")
```

Comment:
check_taint_on_nodes returns 0 even if one of the node doesn't have .OPERATOR_NODE_TAINT. 
May be change this log message to "One or more Nodes already have taint "